ddtrace/profiling/collector/_lock.py

```diff
@@ -262,10 +262,10 @@ def _get_lock_call_loc_with_name(self) -> typing.Optional[str]:
             # 3: caller frame
             # And we expect additional frame if WRAPT_C_EXT is False
             if __debug__:
```

Comment:
Most of our end-users don't run the interpreter with -O, meaning that these checks will be obligatory in prod.  Is that desired?

Reply:
Changed to use `DD_PROFILING_ENABLE_ASSERTS`